core/ecschema-editing/src/Differencing/SchemaDiagnosticVisitor.ts

```diff
@@ -59,19 +56,21 @@ interface LookupArgs {
  */
 export class SchemaDiagnosticVisitor {
 
-  private readonly _differenceReport: SchemaDifferences;
+  private readonly _changes: Array<AnySchemaDifference>;
+  private readonly _conflicts: Array<SchemaDifferenceConflict>;
 
-  constructor(differenceReport: SchemaDifferences) {
-    this._differenceReport = differenceReport;
+  constructor(changes: Array<AnySchemaDifference>, conflicts: Array<SchemaDifferenceConflict>) {
+    this._changes = changes;
+    this._conflicts = conflicts;
   }
 
-  private addEntry<T extends AnySchemaDifference>(entry: T): T {
-    this._differenceReport.changes.push(entry);
+  private addEntry<T extends AnySchemaDifference, R=Extract<AnySchemaDifference, Omit<T, "difference">>>(entry: R): R {
```

Comment:
If there are partial differences which do not have a 'difference' property set  then that should be included in the difference types.  Here we're just lying about the contract.  Folks will be able to get a schema difference with an unset difference property even though the contract is that it will always exist.